CORE-7707 Liquibase logging through SLF4J

davidcurrie opened the pull request:
By default, Liquibase sends all logs to stderr as stdout may be used for outputting the generating SQL. The CLI is writing SQL to files though and we'd rather not see "errors" in the install logs as they tend to jump out at customers. This PR adds a custom logger to Liquibase that just forwards through to SLF4J at the expected log levels.

Commit:
CORE-7707 Liquibase logging through slf4j

chrisr3 commented on tools/plugins/db-config/src/main/kotlin/net/corda/cli/plugins/dbconfig/Slf4jLogger.kt:
```diff
@@ -0,0 +1,90 @@
+    }
+
+    override fun severe(message: String?) {
+        if (logger.isErrorEnabled) {
```
Not to be pedantic, but doesn't SLF4J  _implicitly_ perform these `isErrorEnabled`, `isWarnEnabled` etc checks before logging the message?

davidcurrie replied:
I'll remove them. This was taken from an example elsewhere (and converted to Kotlin) but the original was calling `filterMessage` before delegating the call. That method is now deprecated (it does nothing) so there is no additional work being performed in this wrapper and, as you say, the check will happen at the next level down anyway.

Commit:
CORE-7707 Remove logger checks

The original code was calling filterMessage before delegating to the
logger but that method is now a no-op so the checks are no longer
worthwhile.

chrisr3 commented on tools/plugins/db-config/src/main/kotlin/net/corda/cli/plugins/dbconfig/Slf4jLogService.kt:
```diff
@@ -0,0 +1,11 @@
+
+class Slf4jLogService : AbstractLogService() {
+
+    override fun getPriority() = 5
```
What is "Priority 5"?

davidcurrie replied:
🤷 It just has to be greater than 1 which is the priority of the default logger. In theory, this would allow you to insert additional loggers in between but then, why would you?!

chrisr3 replied:
> shrug It just has to be greater than 1 which is the priority of the default logger. In theory, this would allow you to insert additional loggers in between but then, why would you?!

Well, in that case why not have `DEFAULT_PRIORITY + 1`? Because "5" is just :sparkles: _**MAGIC**_ :sparkles:.

davidcurrie replied:
Switched it to use one of the provided constants (now 10 rather than 5).

Commit:
CORE-7707 Use PRIORITY_SPECIALIZED

chrisr3 reviewed (approved):
LGTM - assuming it doesn't explode :wink:
